quic/s2n-quic-core/src/slice.rs

```diff
@@ -127,15 +124,43 @@ mod tests {
             let mut to = vec![vec![0; 2]; len];
             let copied_len = vectored_copy(&from, &mut to);
             assert_eq!(copied_len, len * 2);
-            assert_eq!(assert_eq_slices(&from, &to), len * 2);
+            assert_eq_slices(&from, &to);
         }
     }
 
-    #[cfg(not(miri))] // bolero doesn't currently work with miri
-    #[test]
+    #[derive(Clone, Copy, Debug, TypeGenerator)]
+    struct InlineVec<T, const LEN: usize> {
+        values: [T; LEN],
+
+        #[generator(_code = "0..LEN")]
+        len: usize,
+    }
+
+    impl<T, const LEN: usize> core::ops::Deref for InlineVec<T, LEN> {
+        type Target = [T];
+
+        fn deref(&self) -> &Self::Target {
+            &self.values[..self.len]
+        }
+    }
+
+    impl<T, const LEN: usize> core::ops::DerefMut for InlineVec<T, LEN> {
+        fn deref_mut(&mut self) -> &mut Self::Target {
+            &mut self.values[..self.len]
+        }
+    }
+
+    const LEN: usize = if cfg!(kani) { 2 } else { 32 };
+
+    #[cfg_attr(not(kani), test)]
+    #[cfg_attr(kani, kani::proof)]
+    #[cfg_attr(kani, kani::unwind(5))]
```

Comment:
I confirmed that `#[cfg_attr(kani, kani::proof, kani::unwind(5))]` works, so the two `cfg_attr` attributes can be merged.